src/bitmap/mutable.rs

```diff
@@ -6,13 +6,19 @@ use crate::{buffer::MutableBuffer, trusted_len::TrustedLen};
 use super::utils::{count_zeros, fmt, get_bit, set, set_bit, BitmapIter};
 use super::Bitmap;
 
+const PADDED_LEN: usize = 64;
+
 /// A container to store booleans. [`MutableBitmap`] is semantically equivalent
 /// to [`Vec<bool>`], but each value is stored as a single bit, thereby achieving a compression of 8x.
 /// This container is the counterpart of [`MutableBuffer`] for boolean values.
 /// [`MutableBitmap`] can be converted to a [`Bitmap`] at `O(1)`.
 /// The main difference against [`Vec<bool>`] is that a bitmap cannot be represented as `&[bool]`.
 /// # Implementation
 /// This container is backed by [`MutableBuffer<u8>`].
+///
+/// # Safety
+/// This struct has the invariant that bits after the current lenght up to an alignment of `[PADDED_LEN]` bytes
+/// must be padded with bits set to 1.
```

Comment:
> 1. reduce the number of "pushes" to the internal buffer when calling bitmap.push by allocating larger regions upfront.
> 2. allocate "set" bits under the assumption that most pushes will be to set them anyway.

Yes, and both have a bit of a win. But we also see a perf increase with 50% nulls, so only point 1. also has merit.

> I think that the idiomatic API to push multiple items to a container is to use an iterator together with the Extend trait (or extend_trusted API in arrow2). PADDED_LEN is an heuristic to address the (assumed) behavior of pushing at least PADDED_LEN items in the future,

I agree. I don't think this may hurt those performances, and the potential reallocs for instance are not acceptable IMO.

The goal was to improve performance of the behavior when you build from rows. In polars, this is the case for instance with csv parsing and when creating from rows (obviously).

I don't like that the assumption leads to a new invariant on the struct, and basically makes it a lot more complicated. I think I'd rather isolate it to a new `unsafe fn push_padded` that documents this assumption. That would isolate the complexity much more.